### PR TITLE
#63 Inheritance commit

### PR DESCRIPTION
**Pull request by ivelin980**

### CODE REVIEW

File: javaOOP_inheritance/src/jv/oop/inheritance/exercise/animals/AnimalsFactoryInt.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package jv.oop.inheritance.exercise.animals;
+
+public interface AnimalsFactoryInt {
```
Comment (ngavrailov) [inline javaOOP_inheritance/src/jv/oop/inheritance/exercise/animals/AnimalsFactoryInt.java:3]: I would name the interface as AnimalsFactory and the implementation class as AnimalsFactoryImpl. But it doesn't matter so much.

**Review by ngavrailov — CHANGES_REQUESTED** [review]

File: javaOOP_inheritance/src/jv/oop/inheritance/exercise/needforspeed/Vehicle.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	}
+
+	private void setHorsePower(int horsePower) {
+		this.horsePower = horsePower;
+	}
+
+	public void drive(double kilometers) {
+		double consumption = getFuelConsumption() * kilometers;
+		if (getFuel() < consumption) {
+			throw new IllegalStateException("The fuel of" + getFuel() + " is not enough");
```
[the remark this answers is not in the document]
Comment (ivelin980) [reply javaOOP_inheritance/src/jv/oop/inheritance/exercise/needforspeed/Vehicle.java:42]: fixed

File: javaOOP_inheritance/src/jv/oop/inheritance/lab/random/arraylist/RandomArrayList.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package jv.oop.inheritance.lab.random.arraylist;
+
+import java.util.ArrayList;
+import java.util.Random;
+
+@SuppressWarnings("serial")
```
[the remark this answers is not in the document]
Comment (ivelin980) [reply javaOOP_inheritance/src/jv/oop/inheritance/lab/random/arraylist/RandomArrayList.java:6]: fixed

File: javaOOP_inheritance/src/jv/oop/inheritance/exercise/animals/Main.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+			try {
+				Animal animal = factory.createAnimal(type, name, age, gender);
+				animals.add(animal);
+			} catch (Exception e) {
+				System.out.println(e.getMessage());
+			}
+			typeInput = scan.nextLine();
+		}
+
+		for (Animal animal : animals) {
```
[the remark this answers is not in the document]
Comment (ivelin980) [reply javaOOP_inheritance/src/jv/oop/inheritance/exercise/animals/Main.java:57]: fixed

File: javaOOP_inheritance/src/jv/oop/inheritance/exercise/animals/Main.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
+package jv.oop.inheritance.exercise.animals;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+/**
```
[the remark this answers is not in the document]
Comment (ivelin980) [reply javaOOP_inheritance/src/jv/oop/inheritance/exercise/animals/Main.java:7]: thanks! :)

File: javaOOP_inheritance/src/jv/oop/inheritance/exercise/animals/AnimalsFactoryInt.java (hunk 1 above)
Comment (ivelin980) [reply javaOOP_inheritance/src/jv/oop/inheritance/exercise/animals/AnimalsFactoryInt.java:3]: Okay

**Review by ngavrailov — APPROVED** [review]

**Pull request merged**